### ISSUE TITLE
Add a new rule for pipes in "RUN"

### ISSUE THREAD

**Issue by zemanlx** [hadolint/hadolint#170]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

Add a new rule for pipes in `RUN`.

> https://github.com/docker/docker.github.io/blob/master/develop/develop-images/dockerfile_best-practices.md#using-pipes
#### Using pipes

Some `RUN` commands depend on the ability to pipe the output of one command into another, using the pipe character (`|`), as in the following example:

```Dockerfile
RUN wget -O - https://some.site | wc -l > /number
```

Docker executes these commands using the `/bin/sh -c` interpreter, which
only evaluates the exit code of the last operation in the pipe to determine
success. In the example above this build step succeeds and produces a new
image so long as the `wc -l` command succeeds, even if the `wget` command
fails.

If you want the command to fail due to an error at any stage in the pipe,
prepend `set -o pipefail &&` to ensure that an unexpected error prevents
the build from inadvertently succeeding. For example:

```Dockerfile
RUN set -o pipefail && wget -O - https://some.site | wc -l > /number
```

> **Note**: Not all shells support the `-o pipefail` option. In such
> cases (such as the `dash` shell, which is the default shell on
> Debian-based images), consider using the *exec* form of `RUN`
> to explicitly choose a shell that does support the `pipefail` option.
> For example:
>

```Dockerfile
RUN ["/bin/bash", "-c", "set -o pipefail && wget -O - https://some.site | wc -l > /number"]
```

**Comment by lorenzo** [comment]
Looks like a nice new rule to have, @zemanlx do you want to take care of implementing it?

**Comment by zemanlx** [comment]
@lorenzo yes, I will, probably next week if anybody else won't pick it up :wink:
